src/Cake.Web/App_Data/blog/2016-05-26-debug-cake-file.md

```diff
@@ -0,0 +1,55 @@
+
+This means that while creating/modifying your Cake build script, you can use Visual Studio to step into the Cake file and get full debug support, in the same way as if you were writing something like a Windows Forms Application.  This includes things like:
+
+* Breakpoints
+* Watches
+* Immediate Window
+* etc.
+
+### How does this work?
+
+In order to enable debugging of a Cake file, follow these steps:
+
+1. Launch cake.exe with `-debug` flag
+1. Process will stop prior to executing script and report `Attach debugger to process xxx to continue`
+<br/>![Pass Debug Flag](https://raw.githubusercontent.com/cake-build/website/master/src/Cake.Web/App_Data/images/debugging-cake-file/pass_debug_flag.png)
+1. Open Visual Studio, open the cake file, add a breakpoint
+1. Open Debug > Attach to Process...
+<br/>![Attach To Process](https://raw.githubusercontent.com/cake-build/website/master/src/Cake.Web/App_Data/images/debugging-cake-file/attach_to_process.png)
+1. Attach to process with PID reported in step 2.
+<br/>![Find Process Id](https://raw.githubusercontent.com/cake-build/website/master/src/Cake.Web/App_Data/images/debugging-cake-file/find_process_id.png)
+1. The execution of the Cake script will then continue, and your breakpoints will be hit
+<br/>![Hit Breakpoint](https://raw.githubusercontent.com/cake-build/website/master/src/Cake.Web/App_Data/images/debugging-cake-file/hit_breakpoint.png)
+
+### Alternative Approach
+
+Another way to enable debugging of a Cake script is to directly debug the Cake Source Code, passing in the location of the Cake file that you would like to debug.  This can be done using the following steps:
+
+1. Clone the Cake project from GitHub
+1. Open the Cake solution file `src/Cake.sln`
+1. Modify the project properties to specify the build.cake file that you would like to debug
+<br/>![Hit Breakpoint](https://raw.githubusercontent.com/cake-build/website/master/src/Cake.Web/App_Data/images/debugging-cake-file/project_properties.png)
+1. Start debugging by pushing F5, or clicking the green play button
+
+### break preprocessor directive
+
+In addition to directly setting breakpoints, it is also possible to place a `#break` pre-processor directive at the location where you want the debugger to break in your script.  At runtime, this `#break` directive will be replaced by `System.Diagnostics.Debugger.Break();` which will cause the debugger to stop at that location.
+
+![Using Break Directive](https://raw.githubusercontent.com/cake-build/website/master/src/Cake.Web/App_Data/images/debugging-cake-file/using_break_directive.png)
+
+### Gotchas!
+
+1. In order to allow debugging of Cake Scripts, Visual Studio must be running in an elevated mode.  If you attempt to debug a Cake script when not elevated, you will see the following message appear.
+<br/>![Elevated Permissions](https://raw.githubusercontent.com/cake-build/website/master/src/Cake.Web/App_Data/images/debugging-cake-file/elevated_permissions.png)
```

Comment:
Now when I think about it, that has never appeared to me as well. Can we change it to something like "Visual Studio might need to run in elevated mode." or something similar?